src/ring_theory/dedekind_domain.lean

```diff
@@ -585,18 +589,92 @@ theorem is_dedekind_domain_iff_is_dedekind_domain_inv :
   is_dedekind_domain A ↔ is_dedekind_domain_inv A :=
 ⟨λ h I hI, by exactI fractional_ideal.mul_inv_cancel hI, λ h, h.is_dedekind_domain⟩
 
-noncomputable instance fractional_ideal.comm_group_with_zero
-  [is_dedekind_domain A] : comm_group_with_zero (fractional_ideal A⁰ K) :=
+end inverse
+
+section is_dedekind_domain
+
+variables {R A} [is_dedekind_domain A] [algebra A K] [is_fraction_ring A K]
+
+open fractional_ideal
+
+noncomputable instance fractional_ideal.comm_group_with_zero :
+  comm_group_with_zero (fractional_ideal A⁰ K) :=
 { inv := λ I, I⁻¹,
   inv_zero := inv_zero' _,
   exists_pair_ne := ⟨0, 1, (coe_to_fractional_ideal_injective (le_refl _)).ne
     (by simpa using @zero_ne_one (ideal A) _ _)⟩,
   mul_inv_cancel := λ I, fractional_ideal.mul_inv_cancel,
   .. fractional_ideal.comm_semiring }
 
-noncomputable instance ideal.comm_cancel_monoid_with_zero
-  [is_dedekind_domain A] : comm_cancel_monoid_with_zero (ideal A) :=
+noncomputable instance ideal.comm_cancel_monoid_with_zero :
+  comm_cancel_monoid_with_zero (ideal A) :=
 function.injective.comm_cancel_monoid_with_zero (coe_ideal_hom A⁰ (fraction_ring A))
   coe_ideal_injective (ring_hom.map_zero _) (ring_hom.map_one _) (ring_hom.map_mul _)
 
-end inverse
+/-- If `I` divides `J`, then `I` contans `J`.
+
+In a Dedekind domain, to divide and contain are equivalent, see `ideal.dvd_iff_le`.
+-/
+lemma ideal.le_of_dvd {I J : ideal R} (hIJ : I ∣ J) : J ≤ I :=
```

Comment:
Why do you prove this one in a separate lemma, but not the other direction? The proof is short enough to inline, I would guess.

Reply:
This is holds for any `comm_ring R`; the converse is specific to Dedekind domains. In fact, I was intending to move this to `ideal/operations.lean`, one moment...